engine/mpm_solver.py

```diff
@@ -7,7 +7,7 @@
 
 USE_IN_BLENDER = False
 
-ti.require_version(0, 7, 10)
+ti.require_version(0, 7, 21)
```

Comment:
```suggestion
ti.require_version(0, 7, 30)
```
?

Reply:
update to 0.7.31 after [this](https://github.com/taichi-dev/taichi/pull/2826)(ti.rescale_index) is merged in `Taichi` repo